### PR TITLE
Create new OpenRCT2 news item struct

### PR DESCRIPTION
**Pull request by IntelOrca**
Same deal as #5076 but for the scores.dat structs and `rct_news_item`.

### CODE REVIEW

File: src/openrct2/management/news_item.h [hunk 1]
```diff
@@ -39,27 +39,22 @@ enum {
...
-assert_struct_size(rct_news_item, 12 + 256);
-#pragma pack(pop)
+typedef struct NewsItem {
+	uint8   Type;
+	uint8   Flags;
+	uint32  Assoc;
+	uint16  Ticks;
+	uint16  MonthYear;
+	uint8   Day;
+	utf8    Text[256];
```
Comment (marijnvdwerf) [inline src/openrct2/management/news_item.h]: Looks like there is some stuff that depends on the length of the text. Might be a good time to `#define` it?

Comment (IntelOrca) [inline src/openrct2/management/news_item.h]: I'd rather remove the dependencies, allow news items to be of any text length and then trimmed as it's passed to other things.

File: src/openrct2/management/news_item.h [hunk 2]
```diff
@@ -39,27 +39,22 @@ enum {
...
-	uint16 month_year;			// 0x08
-	uint8 day;					// 0x0A
-	uint8 pad_0B;				// 0x0B
-	utf8 text[256];				// 0x0C
-} rct_news_item;
-assert_struct_size(rct_news_item, 12 + 256);
-#pragma pack(pop)
+typedef struct NewsItem {
+	uint8   Type;
+	uint8   Flags;
```
Comment (marijnvdwerf) [inline src/openrct2/management/news_item.h]: Looking through the PR, this seems like a boolean property to me

Comment (IntelOrca) [inline src/openrct2/management/news_item.h]: Fixed

File: src/openrct2/management/news_item.h (hunk 1 above)
Comment (IntelOrca) [inline src/openrct2/management/news_item.h]: Fixed, all usages of text use sizeof(Text)

File: src/openrct2/management/news_item.h (hunk 2 above)
Comment (marijnvdwerf) [inline src/openrct2/management/news_item.h]: I meant it more as a "maybe this should be a boolean" instead of "the flags should be named".

Comment (IntelOrca) [inline src/openrct2/management/news_item.h]: But then you wouldn't be able to have more flags?

Comment (marijnvdwerf) [inline src/openrct2/management/news_item.h]: You could just add another boolean property, right?

Comment (IntelOrca) [inline src/openrct2/management/news_item.h]: That would waste another byte

Comment (marijnvdwerf) [inline src/openrct2/management/news_item.h]: If you'd add `:1`, it wouldn't make a difference, right? Except for being more obvious to use than flags.

Comment (IntelOrca) [inline src/openrct2/management/news_item.h]: We use flags everywhere else? What's the obsession with using a `boolean` now? Is it just so the expressions are simpler? In C++ we can always add a helper method to return a boolean representing the flag.

**Pull request merged**